internal/internal_workflow.go

```diff
@@ -1423,6 +1449,7 @@ func newDecodeFuture(ctx Context, fn interface{}) (Future, Settable) {
 
 // setQueryHandler sets query handler for given queryType.
 func setQueryHandler(ctx Context, queryType string, handler interface{}) error {
+	assertNotInReadOnlyState(ctx)
```

Comment:
I think all of these calls should be the first line of the exported calls, not after the interceptors. I think this makes it clearer that you're catching them all and safer to make sure it's not doing any validation or other code first.

Reply:
yeah good catch